src/wp-includes/class-wp-theme-json.php

```diff
@@ -2317,6 +2343,9 @@ public function merge( $incoming ) {
 					) {
 						_wp_array_set( $this->theme_json, $path, $content );
 					} else {
+						$slugs_node = static::get_default_slugs( $this->theme_json, $node['path'] );
+						$slugs      = array_merge_recursive( $slugs_global, $slugs_node );
```

Comment:
It wasn't _added_, it was just _moved_ here. 😉 
A few lines above we were calling  these 2 on the previous `foreach` loop (used to be line 2281), so they were always running. Instead of always calling them, we moved them here where they are actually used. This way these rather expensive calls run fewer times, only when necessary.